### DISCUSSION TITLE
Renovate exiting when "Repository is not found" error occurs

### DISCUSSION THREAD

**Discussion by synlestidae** [renovatebot/renovate#14781] [Request Help] [answered]
Hi there,

I am running Renovate self-hosted with autodiscover as a kubernetes CronJob. I am using self-hosted BitBucket. During the nightly run Renovate exits early, logging `INFO: Renovate is exiting with a non-zero code due to the following logged errors`. The error is `Repository is not found`. It encounters the error for just *one* repository. 

When I visit the repository in BitBucket it tells me:

_The default branch is set to "refs/heads/master" but does not exist. Contact your project administrator to change the default branch._

It is running as a kubernetes CronJob. It's frustrating because Renovate proceeds through each repository in a certain order. When it restarts, it goes from the top and encounters the error again. The repositories the follow the "bad" repository miss out. Is there a way to get Renovate to recover when it encounters errors like this? I don't want to have to add custom rules for it. 

Thanks!

**Comment by rarkins** [comment, marked as answer]
Renovate intentionally logs that message at the end of the whole run so that you don't miss it. It doesn't mean that Renovate exited *immediately* after the error like you suspect.

**Comment by synlestidae** [comment]
I checked the logs again. It looks like `Repository finished (repository=<BAD_REPOSITORY>)` was logged at 06:42. Then `Renovate is exiting with a non-zero code due to the following logged error` was logged at 07:13. It continued for about 30 minutes. Looks like I misinterpreted the behaviour. Thanks @rarkins

**Reply by rarkins** [reply]
Great, thanks for confirming. At least this "feature" of repeating errors at the end caught your attention as intended even if it temporarily confused!
